### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by cdrage**
- Updates the README
 - Changes the location of the installation document
 - Moves contributing content to development file

### CODE REVIEW

**Comment by cdrage** [comment]
Ready for review @kadel @containscafeine @surajnarwade !

File: README.md [hunk 1]
```diff
@@ -1,77 +1,134 @@
...
 
-## Pre-requisites
-- OpenShift version 3.7.0 and up
+#### Features
 
-To use odo you need access to an OpenShift instance and have OpenShift CLI installed on your local machine (`oc` should be in your $PATH).
+  - **Multiple languages:** Odo supports Node.JS, Ruby, .Net Core, Perl, PHP and Python.
+  - **Speed:** Building your source code *immediately* after saving and deployed to your cluster.
+  - **Reproducible:** Allows for easy reproducibility by using tightly versioned Docker containers for your source code environment.
+  - **Deployability:** Easily deploy a new version, or have Odo automatically build and re-deploy your code on each-change.
```
Comment (surajnarwade) [inline README.md]: s/each-change/each change

File: README.md [hunk 2]
```diff
@@ -1,77 +1,134 @@
...
+  - **Support for multiple components and microservices:** Deploy only what you need. For example, having both a Ruby and a JavaScript application side-by-side.
+  - **Serverless:** No requirement to running a server to automate tasks. Odo talks to OpenShift directly through an API.
```
Comment (surajnarwade) [inline README.md]: `for` running server

File: README.md [hunk 3]
```diff
@@ -1,77 +1,134 @@
...
+### Requirements
 
-Check out our [Getting Started](docs/getting-started.md) guide and get going!
+  - `minishift` or an OpenShift environment 3.7.0+, the best way to deploy a development environment is using [Minishift](https://www.katacoda.com/mjelen/courses/introduction/developing-with-odo).
+  - `oc` If you do not have it, there's an excellent guide on the [OpenShift site](https://www.katacoda.com/mjelen/courses/introduction/developing-with-odo) on how to install the latest client.
 
-## CLI Structure
+### Deploying a Node.js application using Odo
+
+For a quick-tutorial on how `odo` works, follow the instructions below! Otherwise, we have an [excellent Katacoda tutorial](https://www.katacoda.com/mjelen/courses/introduction/developing-with-odo) or a [in-depth getting started guide](/docs/getting-started.md).
```
Comment (surajnarwade) [inline README.md]: quick tutorial

File: README.md [hunk 4]
```diff
@@ -1,77 +1,134 @@
...
-Supported systems are Debian, Ubuntu, Fedora, CentOS and macOS. You can find more information about package repositories in 
-[Advanced installation guide](./docs/advanced-installation-guide.md)
+For dynamic languages such as JavaScript or Ruby, Odo starts with a [builder image](https://github.com/sclorg/s2i-ruby-container), which performs the following steps:
 
-```
-curl -L https://github.com/redhat-developer/odo/raw/master/scripts/install.sh | bash
-```
+  1. Starts the container with the injected source code.
+  2. Transforms the source code into the appropriate runnable setup. For example, installing dependencies such as from a `package.json` or `config.ru` file.
+  3. Commit and deploys the new container behind an http proxy such as Apache.
```
Comment (kadel) [inline README.md]: there is no commit and new container with changes in #408

File: README.md [hunk 5]
```diff
@@ -1,77 +1,134 @@
...
+
+#### Deploying:
+
+Typically, a developer:
+
+1. Deploys an OpenShift cluster
```
Comment (kadel) [inline README.md]: Why Developer needs deploy and OpenShift cluster?

File: README.md [hunk 6]
```diff
@@ -1,77 +1,134 @@
...
+For dynamic languages such as JavaScript or Ruby, Odo starts with a [builder image](https://github.com/sclorg/s2i-ruby-container), which performs the following steps:
 
-```
-curl -L https://github.com/redhat-developer/odo/raw/master/scripts/install.sh | bash
-```
+  1. Starts the container with the injected source code.
+  2. Transforms the source code into the appropriate runnable setup. For example, installing dependencies such as from a `package.json` or `config.ru` file.
+  3. Commit and deploys the new container behind an http proxy such as Apache.
+
+#### Deploying:
```
Comment (kadel) [inline README.md]: I'm a little bit confused with "Deploying" section.
It looks like this is describing regular workflow using `oc`. Why is it called Deploying than? vs the Previous paragraph that that is closer to `odo` workflow is called "Building"?

File: README.md [hunk 7]
```diff
@@ -1,77 +1,134 @@
-# ODO - OpenShift Developer Optimization
+# Odo - OpenShift do
 
-[![Build Status](https://travis-ci.org/redhat-developer/odo.svg?branch=master)](https://travis-ci.org/redhat-developer/odo) [![codecov](https://codecov.io/gh/redhat-developer/odo/branch/master/graph/badge.svg)](https://codecov.io/gh/redhat-developer/odo)
+[![Build Status](https://travis-ci.org/redhat-developer/odo.svg?branch=master)](https://travis-ci.org/redhat-developer/odo) [![codecov](https://codecov.io/gh/redhat-developer/odo/branch/master/graph/badge.svg)](https://codecov.io/gh/redhat-developer/odo) [![GoDoc Widget]][GoDoc]
```
Comment (kadel) [inline README.md]: I'm not sure if it is a good idea to add a link to Godoc reference. The output of godoc is pretty useless right now :-(

**Comment by cdrage** [comment]
@kadel @surajnarwade Updated the PR with your comments! Thanks for pointing them all out.

I removed the GoDoc as well as changed from "Deploying" to "Development"

File: README.md [hunk 8]
```diff
@@ -1,77 +1,134 @@
...
+For dynamic languages such as JavaScript or Ruby, Odo starts with a [builder image](https://github.com/sclorg/s2i-ruby-container), which performs the following steps:
 
-```
-curl -L https://github.com/redhat-developer/odo/raw/master/scripts/install.sh | bash
-```
+  1. Starts the container with the injected source code.
+  2. Transforms the source code into the appropriate runnable setup. For example, installing dependencies such as from a `package.json` or `config.ru` file.
+  3. Builds and deploys the new container behind an http proxy such as Apache.
+
+#### Development:
```
Comment (kadel) [inline README.md]: I'm still confused about "Building" and "Development" headers/sections. Why is it called like that?

**Comment by cdrage** [comment]
@kadel Updated the sections. I've changed it to: "Deploying your source code" and "Compiling your source code".

These sections outline the process that developers usually take to get their changes / code out there. The README.md emphasizes that Odo automates (most) of this.

**Pull request merged**